bot/exts/info/information.py

```diff
@@ -34,47 +34,63 @@ def __init__(self, bot: Bot):
         self.bot = bot
 
     @staticmethod
-    def role_can_read(channel: GuildChannel, role: Role) -> bool:
-        """Return True if `role` can read messages in `channel`."""
-        overwrites = channel.overwrites_for(role)
-        return overwrites.read_messages is True
-
-    def get_staff_channel_count(self, guild: Guild) -> int:
-        """
-        Get the number of channels that are staff-only.
+    def is_staff_channel(guild: Guild, channel: GuildChannel) -> bool:
+        """Determines if a given channel is staff-only."""
+        if channel.type is ChannelType.category:
+            return False
+
+        # Channel is staff-only if staff have explicit read allow perms
+        # and @everyone has explicit read deny perms
+        return any(
+            channel.overwrites_for(guild.get_role(staff_role)).read_messages is True
+            and channel.overwrites_for(guild.default_role).read_messages is False
```

Comment:
Not against verbosity, but just making sure it was intended, and not be
```suggestion
            channel.overwrites_for(guild.get_role(staff_role)).read_messages
            and not channel.overwrites_for(guild.default_role).read_messages
```

Reply:
[`overwrites_for`](https://discordpy.readthedocs.io/en/latest/api.html?highlight=overwrites_for#discord.TextChannel.overwrites_for) returns a [`PermissionOverwrite`](https://discordpy.readthedocs.io/en/latest/api.html?highlight=overwrites_for#discord.PermissionOverwrite) object.
> Unlike a regular Permissions, the default value of a permission is equivalent to None and not False. Setting a value to False is explicitly denying that permission, while setting a value to True is explicitly allowing that permission.

An implicit check would also count the `default` permission as a deny, hence we explicitly check for `True` or `False`. I'm not resolving this yet since I think this could be clarified a bit more in the comment.